packages/expo-secure-store/android/src/main/java/expo/modules/securestore/AuthenticationHelper.kt

```diff
@@ -1,212 +1,90 @@
-      )
-    }
-
-    override fun checkAuthentication(
-      promise: Promise,
-      requiresAuthentication: Boolean,
-      cipher: Cipher,
-      gcmParameterSpec: GCMParameterSpec,
-      options: ReadableArguments,
-      encryptionCallback: EncryptionCallback,
-      postEncryptionCallback: PostEncryptionCallback?
-    ) {
-      if (requiresAuthentication) {
-        openAuthenticationPrompt(promise, options, encryptionCallback, cipher, gcmParameterSpec, postEncryptionCallback)
-      } else {
-        handleEncryptionCallback(promise, encryptionCallback, cipher, gcmParameterSpec, postEncryptionCallback)
-      }
+  suspend fun authenticateCipher(cipher: Cipher, requiresAuthentication: Boolean, title: String): Cipher {
+    if (requiresAuthentication) {
+      return openAuthenticationPrompt(cipher, title).cryptoObject?.cipher
+        ?: throw AuthenticationException("Couldn't get cipher from authentication result")
     }
+    return cipher
   }
 
-  fun handleEncryptionCallback(
-    promise: Promise,
-    encryptionCallback: EncryptionCallback,
+  private suspend fun openAuthenticationPrompt(
     cipher: Cipher,
-    gcmParameterSpec: GCMParameterSpec,
-    postEncryptionCallback: PostEncryptionCallback?
-  ) {
-    try {
-      encryptionCallback.run(promise, cipher, gcmParameterSpec, postEncryptionCallback)
-    } catch (exception: GeneralSecurityException) {
-      Log.w(SecureStoreModule.TAG, exception)
-      promise.reject(
-        "ERR_SECURESTORE_ENCRYPT_FAILURE",
-        "Could not encrypt/decrypt the value for SecureStore",
-        exception
-      )
-    } catch (exception: JSONException) {
-      Log.w(SecureStoreModule.TAG, exception)
-      promise.reject(
-        "ERR_SECURESTORE_ENCODE_FAILURE",
-        "Could not create an encrypted JSON item for SecureStore",
-        exception
-      )
-    }
-  }
-
-  private fun openAuthenticationPrompt(
-    promise: Promise,
-    options: ReadableArguments,
-    encryptionCallback: EncryptionCallback,
-    cipher: Cipher,
-    gcmParameterSpec: GCMParameterSpec,
-    postEncryptionCallback: PostEncryptionCallback?
-  ) {
+    title: String
+  ): BiometricPrompt.AuthenticationResult {
     if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
-      promise.reject(
-        "ERR_SECURESTORE_AUTH_NOT_AVAILABLE",
-        "Biometric authentication requires Android API 23"
-      )
-      return
+      throw AuthenticationException("Biometric authentication requires Android API 23")
     }
     if (isAuthenticating) {
-      promise.reject(
-        "ERR_SECURESTORE_AUTH_IN_PROGRESS",
-        "Authentication is already in progress"
-      )
-      return
+      throw AuthenticationException("Authentication is already in progress")
+    }
+
+    isAuthenticating = true
+
+    assertBiometricsSupport()
+    val fragmentActivity = getCurrentActivity() as FragmentActivity?
```

Comment:
```suggestion
    val fragmentActivity = getCurrentActivity() as? FragmentActivity
```
the `as?` is safer, and you are handling errors on your own, so the app should not crash when `getCurrentActivity` cannot be cast.